polars/lazy/expr/index.ts

```diff
@@ -743,6 +743,151 @@ export interface Expr
+  * ```
+  * Passing a mapping with replacements is also supported as syntactic sugar.
+    Specify a default to set all values that were not matched.
+  * ```
+    >>> mapping = {2: 100, 3: 200}
+    >>> df.withColumns(pl.col("a").replace(mapping, default=-1).alias("replaced"))
+    shape: (4, 2)
+    ┌─────┬──────────┐
+    │ a   ┆ replaced │
+    │ --- ┆ ---      │
+    │ i64 ┆ i64      │
+    ╞═════╪══════════╡
+    │ 1   ┆ -1       │
+    │ 2   ┆ 100      │
+    │ 2   ┆ 100      │
+    │ 3   ┆ 200      │
+    └─────┴──────────┘
+  * ```
+    Replacing by values of a different data type sets the return type based on
+    a combination of the `new` data type and either the original data type or the
+    default data type if it was set.
+  * ```
+    >>> df = pl.DataFrame({"a": ["x", "y", "z"]})
+    >>> mapping = {"x": 1, "y": 2, "z": 3}
+    >>> df.withColumns(pl.col("a").replace(mapping).alias("replaced"))
+    shape: (3, 2)
+    ┌─────┬──────────┐
+    │ a   ┆ replaced │
+    │ --- ┆ ---      │
+    │ str ┆ str      │
+    ╞═════╪══════════╡
+    │ x   ┆ 1        │
+    │ y   ┆ 2        │
+    │ z   ┆ 3        │
+    └─────┴──────────┘
+    >>> df.withColumns(pl.col("a").replace(mapping, default=None).alias("replaced"))
+    shape: (3, 2)
+    ┌─────┬──────────┐
+    │ a   ┆ replaced │
+    │ --- ┆ ---      │
+    │ str ┆ i64      │
+    ╞═════╪══════════╡
+    │ x   ┆ 1        │
+    │ y   ┆ 2        │
+    │ z   ┆ 3        │
+    └─────┴──────────┘
+  * ```
+    Set the `return_dtype` parameter to control the resulting data type directly.
+  * ```
+    >>> df.withColumns(pl.col("a").replace(mapping, returnDtype=pl.UInt8).alias("replaced"))
+    shape: (3, 2)
+    ┌─────┬──────────┐
+    │ a   ┆ replaced │
+    │ --- ┆ ---      │
+    │ str ┆ u8       │
+    ╞═════╪══════════╡
+    │ x   ┆ 1        │
+    │ y   ┆ 2        │
+    │ z   ┆ 3        │
+    └─────┴──────────┘
+  * ```
+  * Expression input is supported for all parameters.
+  * ```
+    >>> df = pl.DataFrame({"a": [1, 2, 2, 3], "b": [1.5, 2.5, 5.0, 1.0]})
+    >>> df.withColumns(
+    ...         pl.col("a").replace(
+    ...         old=pl.col("a").max(),
+    ...         _new=pl.col("b").sum(),
+    ...         _default=pl.col("b"),
+    ...     ).alias("replaced")
+    ... )
+    shape: (4, 3)
+    ┌─────┬─────┬──────────┐
+    │ a   ┆ b   ┆ replaced │
+    │ --- ┆ --- ┆ ---      │
+    │ i64 ┆ f64 ┆ f64      │
+    ╞═════╪═════╪══════════╡
+    │ 1   ┆ 1.5 ┆ 1.5      │
+    │ 2   ┆ 2.5 ┆ 2.5      │
+    │ 2   ┆ 5.0 ┆ 5.0      │
+    │ 3   ┆ 1.0 ┆ 10.0     │
+    └─────┴─────┴──────────┘
+   * ``` 
+   */
+  replace(
+    old: Expr | number | number[],
+    _new: Expr | number | number[],
+    _default?: Expr | number | number[],
+    returnDtype?: DataType,
+  ): Expr;
+  replace({
+    old,
+    _new,
+    _default,
+    returnDtype,
+  }: {
+    old: unknown | Expr | number | number[];
+    _new?: Expr | number | number[];
+    _default?: Expr | number | number[];
```

Comment:
why are these all prefixed with  `_`. The `_` prefix usually means that the field is either unused or private. If you are prefixing with `_` because they are reserved words, you should instead suffix them. 

- `new_` instead of `_new`
- `default_` instead of `_default`